modules/common/topology/types.go

```diff
@@ -0,0 +1,33 @@
+/*
+Copyright 2025 Red Hat
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package topology
+
+// TopologyRef - it models a Topology reference and it can be included in the
+// service operators API. It is used to retrieve the referenced Topology
+type TopologyRef struct {
```

Comment:
A service, e.g. `Glance`, can include this in its API with something like:
```golang
// TopologyRef to apply the Policy defined by the associated CR referenced
// by name
TopologyRef *topology.TopologyRef `json:"topologyRef,omitempty"`
```

Reply:
`Error: topology/types.go:21:6: type name will be used as topology.TopologyRef by other packages, and that stutters; consider calling this Ref`

Reply:
@olliewalsh @stuggi renamed the struct to `Topology`  (as done in other modules for Service for example), so we don't mess up w/ `golint`.
Amending my previous comment as now we can include it in services API through:
```golang
// TopologyRef to apply the Policy defined by the associated CR referenced
// by name
TopologyRef *topology.Topology `json:"topologyRef,omitempty"`
```
which was probably the original request change from Ollie.